pkg/openstack/loadbalancer.go

```diff
@@ -1137,6 +1171,58 @@ func (lbaas *LbaasV2) ensureOctaviaListener(lbID string, name string, curListene
 	return listener, nil
 }
 
+func (lbaas *LbaasV2) ensurePrometheusListener(lbID string, name string, curListenerMapping map[listenerKey]*listeners.Listener, _ corev1.ServicePort, svcConf *serviceConfig, _ *corev1.Service) (*listeners.Listener, error) {
```

Comment:
aargh. Now I see that this is a copy paste from the `ensureOctaviaListener`. Maybe it makes sense to apply findings in this func to the `ensureOctaviaListener`. Additionally modify the `ensureOctaviaListener` to support prometheus listener creation. You can replace `, _ *corev1.Service` in the func arguments to `, metrics bool`, since it's not used anyway.